### PR TITLE
Expose Provisioning Request status as events for Pods/Jobs

### PR DESCRIPTION
**Pull request by pajakd**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
In https://github.com/kubernetes-sigs/kueue/pull/2007 the ETA messages from the Provisioning Request are propagated to Workload status. In this PR the objective is to expose these messages as events for job/pod objects. In case of multiple admission checks at some object, the messages are aggregated into a single event. For pod groups, a separate event is recorded for each member of the group.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Updates in admission check messages are recorded as events for jobs/pods. 
```

### CODE REVIEW

File: pkg/controller/jobframework/reconciler.go [hunk 1]
```diff
@@ -451,6 +455,28 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
...
+			if message != "" {
+				message += "; "
+			}
+			message += "Admission check: \"" + check.Name + "\", Message: \"" + check.Message + "\""
+		}
+	}
+	if message != "" {
+		if cJob, isComposable := job.(ComposableJob); isComposable {
+			cJob.ForEach(func(obj runtime.Object) {
+				r.record.Eventf(obj, corev1.EventTypeNormal, ReasonUpdatedWorkload, message)
```
Comment (alculquicondor) [inline pkg/controller/jobframework/reconciler.go:472]: We should have a dedicated reason for this, like `AdmissionCheckUpdates`

Comment (pajakd) [reply pkg/controller/jobframework/reconciler.go]: Done

File: pkg/controller/jobframework/reconciler.go [hunk 2]
```diff
@@ -451,6 +455,28 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
...
 
+func (r *JobReconciler) recordAdmissionCheckUpdate(wl *kueue.Workload, job GenericJob) {
+	message := ""
+	object := job.Object()
+	for _, check := range wl.Status.AdmissionChecks {
+		if check.State == kueue.CheckStatePending && check.Message != "" {
+			if message != "" {
+				message += "; "
+			}
+			message += "Admission check: \"" + check.Name + "\", Message: \"" + check.Message + "\""
```
Comment (alculquicondor) [inline pkg/controller/jobframework/reconciler.go:466]: That's a lot of extra characters.
Maybe just `check.Name: message`

**Review by alculquicondor — COMMENTED** [review]
/approve
Leaving LGTM to @yaroslava-serdiuk

File: pkg/controller/jobframework/reconciler.go (hunk 2 above)
Comment (pajakd) [reply pkg/controller/jobframework/reconciler.go]: Done.

**Pull request merged**